test/Makefile

```diff
@@ -13,15 +13,15 @@ all: testjdis testmem.pass
 testjdis: test.bin
 	awk '/\t/' test.s > test.raw.s
 	../jdis test.bin > test.disassembled.s
-	diff -u test.raw.s test.disassembled.s
+	diff --strip-trailing-cr -u test.raw.s test.disassembled.s
```

Comment:
Shouldn't git take care of translating line endings when checking out test.s, and shouldn't windows' printf("\n") generate a trailing cr? I'm not clear why this is needed.

Reply:
Windows / Visual Studio encodes the next line as 2 bytes: 0xd and 0xa.
The Linux file test.raw.s has an one byte encoded: 0xa

In the case of the test, the jdis tool built with Visual Studio will generate the test.disassembled.s with the 2 bytes.

The option --strip-trailing-cr causes diff to strip the trailing carriage return character before comparing the files.
Without the option, diff will display the differences for all lines and makefile will exit with an error.

Hope it helps.

Reply:
> Are you saying awk is converting them back to unix-style 0xd/'\n'-only when it writes test.raw.s?
You are correct. Except Unix style is 0xa and not 0xd.
On my Windows system, the test.s file has each line ending with 0xd, 0xa.

I have compiled the jdis tool on MSYS2/MinGW32, and test works just fine without the option --strip-trailing-cr.
The tool compiled with Visual Studio requires to have the option to avoid an diff error.